src/NotificationProcessor/RefundNotificationProcessor.php

```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2021 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\NotificationProcessor;
+
+use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionStates;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\System\StateMachine\Exception\IllegalTransitionException;
+
+class RefundNotificationProcessor extends NotificationProcessor implements NotificationProcessorInterface
+{
+    public function process(): void
+    {
+        $context = Context::createDefaultContext();
+        $orderTransaction = $this->getOrder()->getTransactions()->first();
+        $state = $orderTransaction->getStateMachineState()->getTechnicalName();
+        $logContext = [
+            'orderId' => $this->getOrder()->getId(),
+            'orderNumber' => $this->getOrder()->getOrderNumber(),
+            'eventCode' => NotificationEventCodes::REFUND,
+            'originalState' => $state
+        ];
+
+        if ($this->getNotification()->isSuccess()) {
+            $refundedAmount = (int) $this->getNotification()->getAmountValue();
+            $transactionAmount = $this->currencyUtil->sanitize(
+                $orderTransaction->getAmount()->getTotalPrice(),
+                $this->getOrder()->getCurrency()->getIsoCode()
+            );
+
+            if ($refundedAmount > $transactionAmount) {
```

Comment:
Just realised maybe it's also a good idea to add this as an error in the notification record error_message field therefore it's a bit more visible for the merchants what is the problem and with which notification

Action for the merchant can be the following:
Check whether the refund is indeed for the order indicated in the merchantReference. If yes then manually move the order to refunded and check why the value difference

If it's not for that order then it's good that you don't refund it in shopware and then the merchant can reach out to support to figure out why the wrong connection

Reply:
that's a good point, in that case my solution would be to throw an exception here and it will be saved in the [handler](https://github.com/Adyen/adyen-shopware6/blob/df5929d343c71ae92adf751c190ea4d47c29d95f/src/ScheduledTask/ProcessNotificationsHandler.php#L138) as an error on the notification (also logged as an error).

Reply:
perfect :)